### PR TITLE
Standardise A_printf + fix #18 + bump godot ver.

### LINKED ISSUE #18: belmont crashes if no steam path detected
needs graceful handling at least.

### PR DESCRIPTION
**Pull request by int-72h**
these should really be separate PRs but whatever. Belmont still needs a way to take in a custom steam path, + we don't do much validation on this steam path so we need to check the structure.

### CODE REVIEW

File: belmont/project.godot [hunk 1]
```diff
@@ -12,7 +12,7 @@ config_version=5
 
 config/name="Adastral (osprey)"
 run/main_scene="res://boot.tscn"
-config/features=PackedStringArray("4.2", "Mobile")
+config/features=PackedStringArray("4.3", "Mobile")
```
Comment (s1sw) [inline belmont/project.godot:15]: Is the version bump intentional here?

File: winter/Code/headless/headless.cpp [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
-    printf("We couldn't find your sourcemod folder. Please input an alternate sourcemods path.\n");
-  try_again:
-    std::string path;
+  palace* p;
+  try {
+    p = new palace;  // does sanity checks
+  }catch(std::runtime_error &e){
+    printf("We couldn't find your steam folder. Please input an alternate sourcemods path.\n");
+    try_again:
+      std::string path;
```
Comment (s1sw) [inline winter/Code/headless/headless.cpp:14]: Indentation around the label here is inconsistent.

File: winter/Code/headless/headless.cpp [hunk 3]
```diff
@@ -1,22 +1,27 @@
...
     getline(std::cin, path);
     if (!std::filesystem::exists(path)) {
       printf("That's not a valid path. Probably. Try again. Note the folder needs to exist.\n");
       goto try_again;
     }
-    p->sourcemodsPath = std::filesystem::path(path);
+    p = new palace(path);
+  }
+  p->find_sourcemod_path();
+  if (p->sourcemodsPath.empty()) {
```
Comment (s1sw) [inline winter/Code/headless/headless.cpp:23]: If statement with nothing in the body?

File: winter/Code/palace/palace.cpp [hunk 4]
```diff
@@ -234,7 +249,7 @@ int palace::launch_game(const std::string& game_name, const std::string& argumen
       A_error("%s proton version couldn't be found\n", proton_app_name);
       return 1;
     }
-    setenv("STEAM_COMPAT_DATA_PATH", (sys::GetSteamPath() / "steamapps" / "compatdata").string().c_str(), 1);
+    setenv("STEAM_COMPAT_DATA_PATH", (steamPath/ "steamapps" / "compatdata").string().c_str(), 1);
     setenv("STEAM_COMPAT_CLIENT_INSTALL_PATH", sys::GetSteamPath().string().c_str(), 1);
```
Comment (s1sw) [inline winter/Code/palace/palace.cpp:253]: Missed an instance of sys::GetSteamPath

**Review by s1sw — CHANGES_REQUESTED** [review]
Got a few nitpicks, sorry!

File: belmont/project.godot (hunk 1 above)
Comment (int-72h) [reply belmont/project.godot:15]: no but now it is, mentioning in PR

File: winter/Code/headless/headless.cpp (hunk 2 above)
Comment (int-72h) [reply winter/Code/headless/headless.cpp:14]: it's fine for now the whole codebase is a mess (will be fixed in another pr)

File: winter/Code/headless/headless.cpp (hunk 3 above)
Comment (int-72h) [reply winter/Code/headless/headless.cpp:23]: good point

File: winter/Code/palace/palace.cpp (hunk 4 above)
Comment (int-72h) [reply winter/Code/palace/palace.cpp]: fixed in [d302636](https://github.com/AdastralGroup/osprey/pull/23/commits/d302636707110f0a3303075bd39a39aa75c0fc2f)

**Comment by int-72h** [comment]
Thanks for reviewing it! (@s1sw is a friend of mine I asked to review)

**Review by s1sw — APPROVED** [review]
Nitpicks fixed, LGTM!

**Pull request merged**